pkg/generators/forwarding/fluentd/syslog_conf.go

```diff
@@ -1,14 +1,199 @@
 package fluentd
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// The Syslog output fields can be set to an expression of the form $.abc.xyz
+// If an expression is used, its value will be taken from corresponding key in the record
+var keyre = regexp.MustCompile(`^\$(\.[[:word:]]*)+$`)
+
+var tagre = regexp.MustCompile(`\${tag\[-??\d+\]}`)
```

Comment:
Why is this different from payloadRe? I thought they were doing the same thing.

Reply:
`keyre`  takes values from record by jsonpath, and `tagre` takes values from fluentd tag